var/spack/repos/builtin/packages/py-psyclone/package.py

```diff
@@ -7,36 +7,69 @@
 
 
 class PyPsyclone(PythonPackage):
-    """Code generation for the PSyKAl framework from the GungHo project,
-    as used by the LFRic model at the UK Met Office."""
+    """
+    Code generation system developed to support domain-specific languages
+    (DSLs) for finite element, finite volume and finite difference codes.
+    Notably, it is used in the LFRic Project from the UK Met Office, and
+    it also supports the GOcean (2D, finite difference) DSL.
 
+    """
+
+    # Links
     homepage = "https://github.com/stfc/PSyclone"
-    url = "https://github.com/stfc/PSyclone/archive/1.5.1.tar.gz"
     git = "https://github.com/stfc/PSyclone.git"
+    pypi = "PSyclone/PSyclone-2.5.0.tar.gz"
+
+    # License
+    license("BSD-3-Clause")
 
+    # Releases
     version("develop", branch="master")
-    version("1.5.1", commit="eba7a097175b02f75dec70616cf267b7b3170d78")
+    version("2.5.0", sha256="dd1b40d635423c6b23effd2c569908d319afa6153680692e1cbae27f7b5bf4dc")
+    version("2.4.0", sha256="14fd3717f99b317471356c59c1d4c4c22c41fd264af11b78ed831dd2eb71a270")
+    version("2.3.1", sha256="eee70b3069d71fcf95e9bc8796f0333bd502e0202a98df051b635b133432a082")
+    version("2.3.0", sha256="a2cb3f03ad827de99af6acef794d354146443f21623830e4ff62282a81b7cdb3")
+    version("2.2.0", sha256="da829e3b88bf8df7bdb1f261cfc9b20c119eae79fbbd92d970eefee7390ca159")
+    version("2.1.0", sha256="7ef967146d0e2f4662d1d68472242d12f2097adb90646c5646c962ea2e0f187c")
+    version("2.0.0", sha256="94766ffda760404af99f85d70341376192e4a1b8e16e7ae5df980038898a9c41")
 
+    # Current dependencies
     depends_on("py-setuptools", type="build")
     depends_on("py-pyparsing", type=("build", "run"))
+    depends_on("py-configparser", type=("build", "run"))
+    depends_on("py-jsonschema@3.0.2", type=("build", "run"), when="@2.1.0:")
```

Comment:
Where does 3.0.2 come from?

Reply:
From the `setup.py` file for releases 2.1.0 - 2.4.0, `json_schema` had a specified version of 3.0.2 (`2.1.0` on the left and `2.5.0` on the right):

![image](https://github.com/spack/spack/assets/130391369/076ba184-6cf1-4624-b06c-4fee325a748b)